sf-stream/src/failpoint.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_imports)]
+use anyhow::{format_err, Result};
+use aptos_api_types::Error;
+
+#[allow(unused_variables)]
+#[inline]
+pub fn fail_point(name: &str) -> Result<(), Error> {
```

Comment:
What's this for? Logging every error? 

Do we want to use the aptos logger?

Reply:
Failpoints are special things we use to create failures in our code, for testing purposes: https://docs.rs/failpoints/latest/failpoints/

They let us 'simulate' things like failures while sending, receiving, etc- things that would otherwise be very difficult to replicate and test, especially in a hermetic manner :-)